Synchronized course and site modules with updated backend API

Arshiya-rze opened the pull request.

Arshiya-rze committed:
Align course and site modules with updated backend API

FaranakBaran commented on client/src/app/components/courses/course-actions/course-edit/course-edit.component.html:
```diff
@@ -31,94 +30,31 @@
-                @else if (TuitionCtrl.errors && TuitionCtrl.hasError('max')) {
-                <mat-error>حداکثر شهریه ۱۰۰.۰۰۰.۰۰۰ تومان (صد میلیون تومان)
-                    است</mat-error>
+                <mat-label>مدت دوره (دقیقه)</mat-label>
```
مدت کل دوره به ساعت هست نه به دقیقه. مثلا میگیم کلاس full stack 200 ساعته

Arshiya-rze replied:
Faranak jan in az backend minutes hastesh man az roye backend daram mizanam code ro

FaranakBaran replied:
backendesh eshtebahe arshia
to lotfan ino dorost kn, mn baraye sprint baadi in bug ro minevisam ke dorost beshe dar backend

FaranakBaran replied:
یعنی داخل محاسبات backend، داره درست انجام میشه . همون با ساعت محاسبات کلاس رو انجام میده و ساعت کلاس رو در ۶۰ ضرب میکنه که دقیقه اش بدست بیاد ولی اسمش رو اشتباهی گذاشتن نمیدونم  CourseMints یا یه همچین چیزی دقیق یادم نمیاد

FaranakBaran replied:
برای این کامنت، کامیت زدی مهندس؟

Arshiya-rze replied:
Done

FaranakBaran commented on client/src/app/components/courses/course-actions/course-edit/course-edit.component.ts:
```diff
@@ -41,242 +32,83 @@ import { environment } from '../../../../../environments/environment.development
   getCourse(): void {
     if (isPlatformBrowser(this._platformId)) {
       const courseTitle: string | null = this._route.snapshot.paramMap.get('courseTitle');
 
```
استفاده از subscribe به جای snapshot

Arshiya-rze replied:
Done

FaranakBaran commented on client/src/app/components/courses/course-actions/course-edit/course-edit.component.ts:
```diff
@@ -41,242 +32,83 @@ import { environment } from '../../../../../environments/environment.development
 
   updateCourse(): void {
     const courseTitle: string | null = this._route.snapshot.paramMap.get('courseTitle');
 
```
استفاده از subscribe به جای snapshot

Arshiya-rze replied:
Done

FaranakBaran commented on client/src/app/components/courses/course-card/course-card.component.html:
```diff
@@ -1,19 +1,20 @@
-    <span class="text-isStarted-false">هنوز شروع نشده ⏳</span>
+    @if (loggedInUserSig && loggedInUserSig()?.roles?.includes('manager')) {
+    <div class="card-items" [routerLink]="['/dashboard/update-course', courseIn.title]" routerLinkActive
+        #rla="routerLinkActive">
```
تگ  را از "div" به "a" تغییر بده
rla="routerLinkActive"> هیجا استفاده نشده. اگر اضافیه لطفا حذف بشه

Arshiya-rze replied:
Tag az div nemitone taghir peyda bokone be a chon dakhelesh ma details darim va tag haye dgei darim

FaranakBaran commented on client/src/app/components/sites/site-actions/site-create/site-create.component.ts:
```diff
@@ -1,11 +1,91 @@
+    nameCtrl: ['', [Validators.required, Validators.minLength(2), Validators.maxLength(30)]],
+    departmentCtrl: ['', [Validators.required, Validators.minLength(2), Validators.maxLength(30)]],
+    floorCtrl: ['', [Validators.required]], 
+    capacityCtrl: ['', [Validators.required, Validators.min(1)]] 
```
فیلدهای floor و capacity عدد (int) هستند.

Arshiya-rze replied:
Injaro motevah nashodam faranak

FaranakBaran commented on client/src/app/components/sites/site-actions/site-edit/site-edit.component.ts:
```diff
@@ -1,11 +1,121 @@
+  getSite(): void {
+    if (isPlatformBrowser(this._platformId)) {
+      this.siteNameFromRoute = this._route.snapshot.paramMap.get('siteName');
+
```
استفاده از subscribe به جای snapshot

Arshiya-rze replied:
Done

FaranakBaran commented on client/src/app/services/site.service.ts:
```diff
@@ -23,14 +23,22 @@ export class SiteService {
+  }
+
+  getSiteByName(siteName: string): Observable<ShowSite> {
+    return this._http.get<ShowSite>(this._baseApiUrl + 'get-site/' + siteName);
```
استفاده از  "const encodedName = encodeURIComponent(siteName); به جای "siteName"
برای اینکه اگر بین نام سایت فاصله وجود داشت (A 1)، url خراب نشود

Arshiya-rze replied:
Farnak man in ro ham motevajeh nashodm

FaranakBaran replied:
ببین الان این طوری که مثلا ته url یه siteName اضافه کردی، اگر منیجیر اسم سایت رو با فاصله وارد کرده باشه (A 1) این url خراب میشه دیگه. تو url که فاصله معنی نداره

FaranakBaran commented on client/src/app/services/site.service.ts:
```diff
@@ -23,14 +23,22 @@ export class SiteService {
 
-  update(siteUpdate: Partial<SiteUpdate>, targetTitelSite: string) {
-    return this._http.put<Site>(this._baseApiUrl + 'update-site/' + targetTitelSite, siteUpdate);
+  update(siteUpdate: Partial<SiteUpdate>, targetSiteName: string) {
```
const encodedName = encodeURIComponent(targetSiteName);

FaranakBaran commented on client/src/app/services/site.service.ts:
```diff
@@ -23,14 +23,22 @@ export class SiteService {
+  }
+
+  delete(siteName: string): Observable<any> {
+    return this._http.delete(this._baseApiUrl + 'delete-site/' + siteName);
```
const encodedName = encodeURIComponent(siteName);

FaranakBaran commented on client/src/app/models/course.model.ts:
```diff
@@ -1,40 +1,28 @@
+    title: string;
+    description: string;
+    totalMinutes: number; 
+    isActive: boolean;   
```
میتونی به جای ۴ بار تکرار کردن این کد ها این طوری بنویسی:

export interface CourseBase {
    title: string;
    description: string;
    totalMinutes: number;
    isActive: boolean;
}
export interface AddCourse extends CourseBase {}

export interface ShowCourse extends CourseBase {}

export interface CourseUpdate extends CourseBase {}

export interface Course extends CourseBase {
    id: string; }

Arshiya-rze replied:
Done

FaranakBaran commented:
![Uploading IMG_7577.jpeg…]()

نمیتونم لاگین کنم

Arshiya-rze committed:
Refactor: optimize course models structure based on PR feedback

Arshiya-rze committed:
fix(site-service): encode site name in http requests

Using raw strings for site names in URLs caused errors when names contained spaces or special characters (e.g., "Site A").
Added encodeURIComponent to get, update, and delete methods to ensure valid URL formatting.

Arshiya-rze committed:
fix(course-ui): change course duration from minutes to hours

Updated course management components (create, edit, card) and the model to handle and display course duration in hours instead of minutes.
This addresses the PR review feedback stating that courses are defined by total hours (e.g., 200 hours) rather than minutes.

FaranakBaran reviewed (approved):
Done
